src/clj_kondo/impl/analyzer.clj

```diff
@@ -2115,30 +2115,59 @@
                         m)
                       (cons call analyzed))))))))))
 
-(defn lint-keyword-call! [ctx kw namespaced? arg-count expr]
+(defn analyze-keyword-call [ctx kw namespaced? arg-count expr]
   (let [callstack (:callstack ctx)
-        config (:config ctx)]
-    (when-not (config/skip? config :invalid-arity callstack)
-      (let [ns (:ns ctx)
-            ?resolved-ns (if namespaced?
-                           (if-let [kw-ns (namespace kw)]
-                             (or (get (:qualify-ns ns) (symbol kw-ns))
-                                 ;; because we couldn't resolve the namespaced
-                                 ;; keyword, we print it as is
-                                 (str ":" (namespace kw)))
-                             ;; if the keyword is namespace, but there is no
-                             ;; namespace, it's the current ns
-                             (:name ns))
-                           (namespace kw))
-            kw-str (if ?resolved-ns (str ?resolved-ns "/" (name kw))
-                       (str (name kw)))]
-        (when (or (zero? arg-count)
-                  (> arg-count 2))
-          (findings/reg-finding! ctx
-                                 (node->line (:filename ctx) expr :invalid-arity
-                                             (format "keyword :%s is called with %s args but expects 1 or 2"
-                                                     kw-str
-                                                     arg-count))))))))
+        config (:config ctx)
+        ns (:ns ctx)
+        ?resolved-ns (if namespaced?
+                       (if-let [kw-ns (namespace kw)]
+                         (or (get (:qualify-ns ns) (symbol kw-ns))
+                             ;; because we couldn't resolve the namespaced
+                             ;; keyword, we print it as is
+                             (str ":" (namespace kw)))
+                         ;; if the keyword is namespace, but there is no
+                         ;; namespace, it's the current ns
+                         (:name ns))
+                       (namespace kw))
+        kw-str (if ?resolved-ns (str ?resolved-ns "/" (name kw))
+                   (str (name kw)))]
+    (when (and (not (config/skip? config :invalid-arity callstack))
```

Comment:
why was this moved?

Reply:
So we are able to use the existing bindings, or type warnings should not work if invalid arity is skipped?

Reply:
Yeah, if you call the invalid arity, you don't know which overload was called, so we don't emit type warnings then.

Reply:
Still waiting on this one, I think?